js/initcontext.go

```diff
@@ -46,6 +46,7 @@ type InitContext struct {
 
 	// Cache of loaded programs and files.
 	programs map[string]programWithSource
+	exports  map[string]goja.Value
```

Comment:
> Isn't the point of programs to serve as this cache, as the comment says? Why do we need a separate cache for "exports"?

`programs` is only about source modules - the kind that are js code. This works for both the source and go ones ;)

> Also, does this affect [this comment](https://github.com/grafana/k6/blob/5fa71b761185200d29c87eff16faa5e4a6c372b7/js/initcontext.go#L78-L80) at all, since technically exports/modules are shared now?

That comment is about sharing them *between* vus. 

> Ah, I think this is what https://github.com/grafana/k6/pull/2773#issuecomment-1317345783 is referring to, so I take it you're still working on this.

this mostly refers to rewriting this whole mess to be more layered and composed. Unfortunately it is now looking like half way to #2563 (without any of the goja changes and smaller interfaces). And it still has some of the same problems. As well as I am finding more bugs in this codebase :sob: 

So the current PR is "final" as it fixes a particular discrepancy in how modules work and the refactor looks like it will take me at least a couple more days of work - which I am not certain will happen in this release cycle at this point :(